pkg/kubelet/server/server.go

```diff
@@ -240,6 +248,20 @@ func NewServer(
 	resourceAnalyzer stats.ResourceAnalyzer,
 	auth AuthInterface,
 	kubeCfg *kubeletconfiginternal.KubeletConfiguration) Server {
+	if utilfeature.DefaultFeatureGate.Enabled(features.KubeletTracing) {
+		opts := []otlpgrpc.Option{}
+		if kubeCfg.Tracing.Endpoint == nil {
+			opts = append(opts, otlpgrpc.WithEndpoint(*kubeCfg.Tracing.Endpoint))
+		}
+		opts = append(opts, otlpgrpc.WithInsecure())
+		tp, err := traces.InitTracing(context.Background(), host.GetHostname(), kubeletServiceName, *kubeCfg.Tracing.SamplingRatePerMillion, opts)
```

Comment:
Ideally, we should only construct the tracer provider once in the kubelet, and should plumb it through to wherever it is needed (grpc/http handlers)